### PR TITLE
feat(gas-report): create package, move relevant files to it

### PR DESCRIPTION
**Pull request by dk1a**
fixes https://github.com/latticexyz/mud/issues/667

### CODE REVIEW

**Comment by dk1a** [comment]
> thoughts on refactoring gas report TS to be more of a output parser than a test runner?

if you mean to separate it into a parser and a runner that seems nice if not particularly important;
either way I'd rather keep this PR simple, and move on to gas optimizations

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-lockfileVersion: '6.1'
+lockfileVersion: '6.0'
```
Comment (alvrs) [inline pnpm-lock.yaml:1]: which version of pnpm are you currently on?

**Review by alvrs — COMMENTED** [review]
looking good! would you mind adding a simple example for how to use the gas report to the readme and adding a changeset?

File: pnpm-lock.yaml (hunk 1 above)
Comment (dk1a) [reply pnpm-lock.yaml:1]: 8.6.7

**Comment by dk1a** [comment]
> adding a changeset

I'm assuming that during we don't want to strictly follow semver for now and avoid `major`

**Review by holic — APPROVED** [review]

**Pull request merged**